kaldb/src/test/java/com/slack/kaldb/logstore/opensearch/FieldLimitTest.java

```diff
@@ -0,0 +1,32 @@
+package com.slack.kaldb.logstore.opensearch;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.slack.kaldb.metadata.schema.LuceneFieldDef;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.junit.Test;
+
+public class FieldLimitTest {
+
+  @Test
+  public void checkIfMaxFieldLimitsIsEnforced() {
+    Map<String, LuceneFieldDef> fieldDefMap = getChunkSchema(5001);
+    OpenSearchAdapter openSearchAdapter = new OpenSearchAdapter(fieldDefMap);
+    assertThat(openSearchAdapter.tryRegisterFields()).isFalse();
```

Comment:
We may want to slightly change this behavior, such that the first 5k are allowed (and still returns false). Then the following attempt to register 2 would fail because we already have 5k.

If we think of the situation where a cache node has 5001 fields, it would never register any of them instead of just dropping one that's over the limit, which I'm not sure is what we'd want to do.

Reply:
We could, however this `tryRegisterFields` only registers new fields - So in real life within one commit refresh we'd need to have 5k new fields get added?

Reply:
This would be expected of a cache node, where it will have a single commit with all the fields populated at the initial load.

Reply:
So should we build our logic within `tryRegisterFields` to see if the number of new fields is not beyond X?

Reply:
If `tryRegisterFields` is synchronized, we should be able to check the current length and then just attempt to insert however many is safe to add.

Reply:
```
      CompressedXContent content = new CompressedXContent(BytesReference.bytes(createMappings(Map.of("_kaldb", b -> b.field("type", "keyword")))));
      DocumentMapper documentMapper = mapperService.documentMapperParser().parse("_doc", content);
      MappingLookup mappers = documentMapper.mappers();
      AtomicInteger count = new AtomicInteger();
      mappers.iterator().forEachRemaining(c -> count.getAndIncrement());
```

I figured if we do this, we can get the current count of the field defined. This involved staring at the internals quite a bit

Should we add this and then see if  `newFields.length + count> limit` ? 

Or did your comment just refer to ensuring `newFields.length > limit` ?

Reply:
After looking a bit deeper into this, I think we should probably roll with what you have here in this PR and we may circle back on this later. This seems to be a bit more involved than we probably want to do right now, especially considering that it may change with our schema refactoring.